src/error.rs

```diff
@@ -15,10 +19,32 @@ pub enum Error {
     /// The degree is too high for the coefficients to be indexed by `usize`.
     #[error("The degree is too high for the coefficients to be indexed by usize.")]
     DegreeTooHigh,
+    /// An error reading a structure from an array of bytes. Invalid bytes representation.
+    #[error("Invalid bytes representation.")]
+    InvalidBytes,
+    /// BLST error
+    #[error("BLST error: {0}")]
+    BlstError(String),
 }
 
-/// A crypto result.
-pub type Result<T> = ::std::result::Result<T, Error>;
+impl From<PrimeFieldDecodingError> for Error {
+    fn from(_: PrimeFieldDecodingError) -> Self {
+        Error::InvalidBytes
+    }
+}
+
+impl From<BlstError> for Error {
+    fn from(error: BlstError) -> Self {
+        if error == BlstError::BLST_SUCCESS {
+            Error::BlstError(format!(
+                "received inconsistent and unexpected result: {:?}",
```

Comment:
Just checking if this shows the enum index or string, eg from [blst/bindings.rs L16](https://github.com/supranational/blst/blob/17506b6c096a35b570a566dfefd336855c1385dd/bindings/rust/src/bindings.rs#L16) would it show `1` or would it show `BLST_BAD_ENCODING`? Not a big detail but I wanted to double check since the string is much more meaningful.

Reply:
It seems the string is still printed, at least that's what I see here: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9b975c8134faae47e67abb738ac91e6b